pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -1120,10 +1121,7 @@ func (r *Reconciler) determineRolloutStateForPlacementWithExternalRolloutStrateg
 	} else {
 		placementStatus.ObservedResourceIndex = observedResourceIndex
 		// Construct placement key for the resource collection function
-		placementKey := placementObj.GetName()
-		if placementObj.GetNamespace() != "" {
-			placementKey = placementObj.GetNamespace() + "/" + placementObj.GetName()
-		}
+		placementKey := controller.GetObjectKeyFromNamespaceName(placementObj.GetNamespace(), placementObj.GetName())
```

Comment:
For next PR, I think we should change the CollectResourceIdentifiersFromResourceSnapshot  func signature to take in namespacedName instead of string